### ISSUE TITLE
Certain icons are too large

### ISSUE THREAD

**Issue by kwrooijen** [domtronn/all-the-icons.el#6]
Some of the file icons being displayed are too big, here are some screenshots:
https://s17.postimg.io/7yquh6767/Screenshot_from_2016_09_15_14_24_10.png
https://s9.postimg.io/jzuzm0dbz/Screenshot_from_2016_09_15_15_03_15.png

This issue was discussed here as well: https://github.com/hlissner/emacs-doom-theme/issues/8

I'm running Linux (Fedora 25) and Emacs 25.1.1 (but Emacs 24 has the same issue)

I installed the fonts by dropping the ttf files in `/usr/share/fonts/all-the-icons` and running `fc-cache -f -v`.

The icons that I'm aware of that are having this issue are:
* terminal (for example .sh)
* python (.py)
* html (.html .erb)

I believe I installed the fonts correctly as other icons are being displayed properly, other people (but not all) are also having this issue (see emacs-doom-theme link).

If there's any other info I can give you let me know.

PS: as for Emacs configuration, I tested it (as well) with a completely raw emacs config and only install `all-the-icons` through melpa. [image](https://s17.postimg.io/u26d4v73j/Screenshot_from_2016_09_15_16_00_25.png)

**Comment by domtronn** [comment]
Hi @kwrooijen, I've had a quick look and the `.sh` and `.py` icons are from the `icomoon`/`devicon` icon set... I'd be interested to know whether this happens for all of the icons in that iconset.

Could you try inserting or checking the icon for any of the following file types?
`(insert (all-the-icons-devicon "java"))` a `.java` file
`(insert (all-the-icons-devicon "sass"))` a `.scss` file
`(insert (all-the-icons-devicon "gulp"))` a `.gulpfile`

I have no idea why this font would cause the problem. I'm running OSX don't have this problem, and I'm not sure how fonts are installed on or packaged for Linux... If this does happen for the above files as well, I would assume it's a problem with that Iconset!

**Comment by kwrooijen** [comment]
Hi @domtronn 

Just tested all 3 (except ^gulpfile instead of .gulpfile but I assume that's what you meant).
And they are all too large, so I think that you are correct that there is something wrong with the iconset.

**Comment by kwrooijen** [comment]
Seems like the project is abandoned? https://github.com/vorillaz/devicons/
Not sure what we can really do about this, I have next to none experience with fonts :)

**Comment by kwrooijen** [comment]
Unless you have a better idea, would you be ok if I replaced any devicons I can with any of the other fonts? I could make a PR for that, I'll leave any devicons which can't be replaced.

If you agree with this, it would be nice to add an option to disable any devicons and just use the default file icon. That way people with this issue can disable them.

Your thoughts?

**Comment by domtronn** [comment]
Thanks for testing that - yeah, does seem to be the icon set... Does look like the project is no longer maintained. I was thinking that I could take all of the icons in that set, and add them to my custom font, essentially replacing the `devicon` set with my `alltheicon` set, however that's probably going to take some time to port all of them across. I'll probably be able to find some time to do this over the weekend 👍 

In the mean time, finding icon alternatives for the `devicon` icons seems sensible if you'd like to make a PR, unless you're happy to wait a couple of days :)

**Comment by domtronn** [comment]
@kwrooijen, would you mind just testing that a

```elisp
(insert (all-the-icons-alltheicon "perl")) ;; a .pl file
```

Works as you expect?

**Comment by kwrooijen** [comment]
That one seems to be working fine. Is that a ported icon by any chance?

**Comment by domtronn** [comment]
Not exactly, but it's an icon in an set that I created myself so this means my method of creating the Icon Fonts works correctly on Linux, so porting them _should_ work!

**Comment by kwrooijen** [comment]
All right, good to know :) I can definitely wait. I'll just remove the devicons so that I don't have overly large icons in my neotree. If there is anything else I can help with let me know.

**Comment by domtronn** [comment]
So I've now ported all of the `devicons` icons across into the new/merged `all-the-icons.ttf`. Would you mind installing this and trying out the icons you had problems with before? Hopefully the heights should now work correctly! _(fingers crossed anyway)_

It's worth noting whilst I was testing this, I had some problems on OSX with the Fonts being cached... I had to run the following command

```
atsutil databases -removeUser
```

To clear the font cache before installing/reinstalling a font... Not sure if you'll have to do this as the font name is new! But if you're having any troubles, try clearing the Font Cache on Linux _(quick google says this `sudo fc-cache -f -v` should clear it)_.

Also, I've added a new function which _should_ help with debugging the fonts :) You can now run the following to print _all_ the icons that are supported by that icon set! 

```elisp
;; (all-the-icons-insert-icons-for 'family-name &optional height)
;; for example

(all-the-icons-insert-icons-for 'alltheicons)
```

Useful to see what's available and to see how they print! I'd recommend running this in a `fundamental-mode` buffer.

**Comment by kwrooijen** [comment]
Seems to be working now!
![image](http://i.imgur.com/jCaXr6D.png)

I think it's safe to say that this issue is resolved. Thanks for the effort :)

**Comment by domtronn** [comment]
No problem :) Glad I could fix it!

**Comment by nlamirault** [comment]
Thanks. I will try that on monday.

**Comment by bashhack** [comment]
Just chiming in to say thanks and confirm fix working here on Ubuntu 16.04. Very much appreciated, @domtronn!

**Comment by nlamirault** [comment]
IT works for me too :) THanks.
